### ISSUE TITLE
useless_format: format_err!("my error") from Failure is valid

### ISSUE THREAD

**Issue by klausi** [rust-lang-nursery/rust-clippy#2615]
The following is detected wrongly by clippy:

```rust
format_err!("path sources must have a valid path")
```
```
warning: useless use of `format!`
   --> src/cargo/ops/cargo_install.rs:172:27
    |
172 |             .map_err(|()| format_err!("path sources must have a valid path"))?;
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using .to_string(): `"path sources must have a valid path".to_string()`
    |
    = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.191/index.html#useless_format
    = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

To reproduce the error you can run clippy on the cargo source https://github.com/rust-lang/cargo , which will emit a lot of false positives of this error.

**Comment by mikerite** [comment]
This is from the failure crate.
```
#[macro_export]
macro_rules! format_err {
    ($($arg:tt)*) => { $crate::err_msg(format!($($arg)*)) }
}
```
Maybe this lint will get a macro check but you could also use:
```
.map_err(|()| failure::err_msg("path sources must have a valid path"))?;
```

**Comment by phansch** [comment]
Yup, looks like a macro check would be the best way to fix this. I will take a look later today.

**Comment by phansch** [comment]
Also: https://github.com/rust-lang-nursery/rust-clippy/issues/1130
